scandium-core/src/main/java/org/eclipse/californium/scandium/auth/PrincipalSerializer.java

```diff
@@ -92,10 +98,16 @@ public static void serialize(final Principal principal, final DatagramWriter wri
 
 	private static void serializeIdentity(final PreSharedKeyIdentity principal, final DatagramWriter writer) {
 		writer.writeByte(ClientAuthenticationType.PSK.code);
-		byte[] virtualHost = principal.getVirtualHost() == null ? EMPTY_BYTE_ARRAY :
-			principal.getVirtualHost().getBytes(StandardCharsets.UTF_8);
-		writeBytesWithLength(PSK_LENGTH_BITS, virtualHost, writer);
-		writeBytesWithLength(PSK_LENGTH_BITS, principal.getIdentity().getBytes(StandardCharsets.UTF_8), writer);
+		if (principal.isScopedIdentity()) {
+			writer.writeByte((byte) 1); // scoped
+			byte[] virtualHost = principal.getVirtualHost() == null ? EMPTY_BYTE_ARRAY
+					: principal.getVirtualHost().getBytes(StandardCharsets.UTF_8);
+			writeBytesWithLength(PSK_LENGTH_BITS, virtualHost, writer);
```

Comment:
it sounds a bit strange to use `PSK_LENGHT_BITS` constant for `virtualHost` length.

Reply:
That's not changed. Should we change that to LENGTH_BITS ?
Or other proposal?

Reply:
It could be,  or adding a `HOST_LENGTH_BITS` (or `VIRTUALHOST_LENGTH_BITS`).

Reply:
PSK_HOSTNAME_LENGTH_BITS and PSK_IDENTITY_LENGTH_BITS are now used